onnxruntime/core/providers/mkldnn/subgraph/subgraph.h

```diff
@@ -0,0 +1,71 @@
+// Copyright(C) 2019 Intel Corporation
+// Licensed under the MIT License
+
+#pragma once
+
+#include <vector>
+#include <string>
+#include <map>
+#include "core/framework/op_node_proto_helper.h"
+#include "core/graph/graph.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+struct MklDnnNode {
+  std::string name;
+  int node_index = -1;
+  int input_start_index = -1;  // start index in inputs()
+  int num_inputs = 0;          // and how many inputs
+  int output_index = -1;       // index in output()
+  std::string weight_name;
+  std::string output_name;
+  std::vector<size_t> parent_nodes; // index to parents in vector mklnodes
+
+  std::string ToString() const {  // For Debug purpose only
+    std::string key;
+    key.reserve(128);
+    key.append(name);
+    key.append(", input_start_index: ");
+    key.append(std::to_string(input_start_index));
+    key.append(",num_inputs: ");
+    key.append(std::to_string(num_inputs));
+    key.append(",output_index: ");
+    key.append(std::to_string(output_index));
+    key.append(",output_name: ");
+    key.append(output_name);
+    key.append(", Parent nodes");
+    for (auto& out : parent_nodes)
+      key.append(std::to_string(out) + ",");
+    key.append(";");
+    return key;
+  }
+};
+
+struct Subgraph {
+  struct SubgraphVariables {
+    std::vector<std::string> inputs;
+    std::vector<std::string> outputs;
+    std::vector<std::string> outputs_as_input_other_node;
+    std::vector<onnxruntime::NodeIndex> subgraph_node_indexes;
+    std::shared_ptr<Subgraph> subgraph_ptr;
```

Comment:
an inner class creates an instance of an outer class. not sure if i'm understanding this relationship.

Reply:
Subgraph::SubgraphVariables a structure used as parameter in GetCapability()

SubgraphVariables struct was outside of Subgraph and in one of the last reviews, it was suggested to move it inside.

Reply:
Why does SubgraphVariables have to create an instance of Subgraph? I was expecting the data structure to be something like this. Would this work?

```
struct Subgraph {
  struct SubgraphVariables {
      ....
     void Reset() {
     }
  }
  Subgraph() : subgraph_variables_.subgraph_index(0) {
  }
  void Reset() {
    subgraph_variables_.Reset();
  }
  SubgraphVariables subgraph_variables_;
  std::string subgraph_id;
  std::vector<MklDnnNode> mkldnn_nodes;
}
```

Reply:
thanks for making this change.